src/components/Signup/Nickname.tsx

```diff
@@ -0,0 +1,280 @@
+import React, { useState } from 'react';
+import { Cookies } from 'react-cookie';
+import { useForm } from 'react-hook-form';
+import { useLocation, useNavigate } from 'react-router-dom';
+import styled, { css } from 'styled-components';
+
+import { IcAfterCheckbox, IcBeforeCheckbox } from '../../asset/icon';
+import { checkDuplicateNickname, postSignupInfo } from '../../lib/api/signup';
+import { AddAccountInfo, NicknameInfo } from '../../types/signup';
+import { nicknameErrorPatterns } from '../../utils/check';
+
+const Nickname = () => {
+  const location = useLocation();
+  const cookies = new Cookies();
+
+  const { user_id, password }: AddAccountInfo = location.state.dataInfo;
+
+  const [isChecked, setIsChecked] = useState<boolean[]>([false, false, false]);
+  const [isDuplicate, setIsDuplicate] = useState(false);
+  const [nickname, setNickname] = useState<string>('');
+
+  const termList: string[] = ['만 14세 이상이에요', '이용약관 및 개인정보수집이용 동의'];
+
+  const navigate = useNavigate();
+
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+  } = useForm<NicknameInfo>({ mode: 'onBlur' });
+
+  const handleCheckNickname = () => {
+    checkDuplicateNickname(nickname).then((result) => {
+      if (result?.success) {
+        setIsDuplicate(!result.success);
+        setNickname(nickname);
+      } else {
+        setIsDuplicate(result.success);
+      }
+    });
+  };
+
+  const handleCheck = (e: React.MouseEvent<HTMLElement>, idx?: number) => {
+    const target = e.target as HTMLInputElement;
+    if (target.name === 'all') {
+      isChecked[0] ? setIsChecked([false, false, false]) : setIsChecked([true, true, true]);
+    } else {
+      if (idx) {
+        isChecked[idx] = !isChecked[idx];
+        isChecked[0] = isChecked[1] && isChecked[2] ? true : false;
+        setIsChecked([...isChecked]);
+      }
+    }
+  };
+
+  const handleSignup = () => {
+    postSignupInfo({ user_id, password }, nickname).then((res) => {
+      if (res?.data.success) {
+        cookies.set('refreshToken', res.data.refreshToken);
+        localStorage.setItem('accessToken', res.data.accessToken);
+      }
+    });
+    navigate('/');
+  };
+
+  return (
+    <>
+      <StWrapper>
```

Comment:
```suggestion
      <StWrapper>
```

어차피 `StWrapper` 가 컴포넌트 전체를 관장하는 최상위 컴포넌트라면 상위에 빈 태그는 없애도 될듯~